src/rust/event-source/src/config.rs

```diff
@@ -0,0 +1,39 @@
+use std::net::SocketAddr;
+
+use clap::Parser;
+
+#[derive(Parser, Clone, Debug)]
+pub struct EventSourceConfig {
+    #[clap(flatten)]
+    pub service_config: EventSourceServiceConfig,
+    #[clap(flatten)]
+    pub db_config: EventSourceDbConfig,
+}
+
+impl EventSourceConfig {
+    /// An alias for clap::parse, so that consumers don't need to
+    /// declare a dependency on clap
+    pub fn from_env_vars() -> Self {
+        Self::parse()
+    }
+}
```

Comment:
fwiw another way to solve this is to have your crate `pub use thing_to_export`. If you do `pub use clap::Parser` the consumer won't have to have clap in their toml. But this is a fine solution too.